src/encoded/static/components/viz/QuickInfoBar.js

```diff
@@ -108,64 +108,29 @@ export default class QuickInfoBar extends React.PureComponent {
             'reallyShow'            : false,
             'togglingBrowseState'   : false
         };
+
+        this.memoized = {
+            expSetFilters: memoize(searchFilters.contextFiltersToExpSetFilters),
+            anyFiltersSet: memoize(QuickInfoBar.anyFiltersSet),
+            isInvisibleForHref: memoize(QuickInfoBar.isInvisibleForHref)
+        };
     }
 
     /** @private */
     componentDidMount(){
         this.setState({ 'mounted' : true });
     }
 
-    /**
-     * Check if QuickInfoBar instance is currently invisible, i.e. according to props.href.
-     *
-     * @public
-     * @instance
-     * @returns {boolean} True if counts are null or on a 'href' is not of a page for which searching or summary is applicable.
-     */
-    isInvisible(props = this.props, state = this.state){
-        var { total, current } = QuickInfoBar.getCountsFromProps(props);
-        if (
-            !state.mounted ||
-            props.invisible ||
-            total === null ||
-            (
-                (current.experiment_sets === null && total.experiment_sets === null) &&
```

Comment:
To simplify all this, we no longer hide QuickInfoBar if barplot_data is `null` (presumed not yet loaded - pretty rare case to begin with) or something; instead we just show something like "0 Experiment Sets"/